src/wayland/compositor/tree.rs

```diff
@@ -306,7 +306,8 @@ impl PrivateSurfaceData {
         }
         my_data
             .pending_transaction
-            .insert_state(surface.clone(), my_data.current_txid);
+            .insert_state(surface.clone(), current_txid);
+        my_data.current_txid.0 = my_data.current_txid.0.wrapping_add(1);
```

Comment:
@elinorbgr @Drakulix @ids1024 Not sure this is correct, but without incrementing the transaction id the blocked state will not be serialized but directly merged into any already blocked transaction. This means that for example the buffer will be overwritten and might get a release making fifo unusable with blockers (same for explicit sync, but I guess there it might not be an issue). It would also break my fifo barrier logic because it would allow to override an already pending barrier.